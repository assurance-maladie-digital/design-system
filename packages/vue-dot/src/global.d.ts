import { VueConstructor } from 'vue';
import { VueDotOptions } from '../types';
import { Framework } from 'vuetify/types';

// Extend global interfaces

/* eslint-disable @typescript-eslint/no-explicit-any */
type UnknownValue = any;

declare global {
	// Add window.Vue
	interface Window {
		Vue?: VueConstructor;
	}

	// @see https://github.com/microsoft/TypeScript/issues/45612
	interface Navigator {
		msSaveBlob(blob: UnknownValue, defaultName?: string): boolean;
		msSaveOrOpenBlob(blob: UnknownValue, defaultName?: string): boolean;
	}
}

declare module 'vue/types/vue' {
	// Add this.$vuetify
	interface Vue {
		$vd: VueDotOptions;
		$vuetify: Framework;
	}
}
