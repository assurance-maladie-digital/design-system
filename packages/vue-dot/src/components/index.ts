import { VueConstructor } from 'vue';

import CopyBtn from './CopyBtn.vue';
import CustomIcon from './CustomIcon.vue';
import DataList from './DataList.vue';
import DatePicker from './DatePicker.vue';
import FileList from './FileList.vue';
import FileUpload from './FileUpload.vue';
import LangBtn from './LangBtn.vue';
import NotificationBar from './NotificationBar.vue';
import PaginatedTable from './PaginatedTable.vue';
import UploadWorkflow from './UploadWorkflow.vue';

interface Components {
	[name: string]: VueConstructor;
}

const components: Components = {
	CopyBtn,
	CustomIcon,
	DataList,
	DatePicker,
	FileList,
	FileUpload,
	LangBtn,
	NotificationBar,
	PaginatedTable,
	UploadWorkflow
};

/** Globally register all components */
export default (Vue: VueConstructor) => {
	Object.keys(components).forEach((name: string) => {
		Vue.component(name, components[name]);
	});
};
