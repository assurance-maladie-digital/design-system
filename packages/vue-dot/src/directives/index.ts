import { debounce } from './debounce';

import { VueConstructor, DirectiveOptions } from 'vue';

interface Directives {
	[key: string]: DirectiveOptions;
}

const directives: Directives = {
	debounce
};

/** Globally register all directives */
export default (Vue: VueConstructor) => {
	Object.keys(directives).forEach((name: string) => {
		Vue.directive(name, directives[name]);
	});
};
