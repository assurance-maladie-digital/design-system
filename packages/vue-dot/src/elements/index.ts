import BackBtn from './BackBtn';
import BackToTopBtn from './BackToTopBtn';
import ChipList from './ChipList';
import CopyBtn from './CopyBtn';
import CustomIcon from './CustomIcon';
import DataList from './DataList';
import DialogBox from './DialogBox';
import DownloadBtn from './DownloadBtn';
import ExternalLinks from './ExternalLinks';
import FileList from './FileList';
import FilePreview from './FilePreview';
import FranceConnectBtn from './FranceConnectBtn';
import HeaderLoading from './HeaderLoading';
import Logo from './Logo';
import LogoBrandSection from './LogoBrandSection';
import PageContainer from './PageContainer';
import SkipLink from './SkipLink';
import UserMenuBtn from './UserMenuBtn';

export const elements = {
	BackBtn,
	BackToTopBtn,
	ChipList,
	CopyBtn,
	CustomIcon,
	DataList,
	DialogBox,
	DownloadBtn,
	ExternalLinks,
	FileList,
	FilePreview,
	FranceConnectBtn,
	HeaderLoading,
	Logo,
	LogoBrandSection,
	PageContainer,
	SkipLink,
	UserMenuBtn
};
