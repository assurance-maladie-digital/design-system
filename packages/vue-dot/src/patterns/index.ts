import AlertWrapper from './AlertWrapper';
import ContextualMenu from './ContextualMenu';
import CookieBanner from './CookieBanner';
import DataListGroup from './DataListGroup';
import DatePicker from './DatePicker';
import FileUpload from './FileUpload';
import FilterModule from './FilterModule';
import FiltersInline from './FiltersInline';
import FiltersSideBar from './FiltersSideBar';
import FooterBar from './FooterBar';
import FooterWrapper from './FooterWrapper';
import HeaderBar from './HeaderBar';
import LangBtn from './LangBtn';
import NirField from './NirField';
import NotificationBar from './NotificationBar';
import PaginatedTable from './PaginatedTable';
import PasswordField from './PasswordField';
import PeriodField from './PeriodField';
import PhoneField from './PhoneField';
import RangeField from './RangeField';
import RatingPicker from './RatingPicker';
import SearchListField from './SearchListField';
import SelectBtnField from './SelectBtnField';
import SubHeader from './SubHeader';
import TableToolbar from './TableToolbar';
import UploadWorkflow from './UploadWorkflow';

export const patterns = {
	AlertWrapper,
	ContextualMenu,
	CookieBanner,
	DataListGroup,
	DatePicker,
	FileUpload,
	FilterModule,
	FiltersInline,
	FiltersSideBar,
	FooterBar,
	FooterWrapper,
	HeaderBar,
	LangBtn,
	NirField,
	NotificationBar,
	PaginatedTable,
	PasswordField,
	PeriodField,
	PhoneField,
	RangeField,
	RatingPicker,
	SearchListField,
	SelectBtnField,
	SubHeader,
	TableToolbar,
	UploadWorkflow
};
