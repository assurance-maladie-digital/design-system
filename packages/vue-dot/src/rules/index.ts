import email, { email as emailFn } from './email';
import isDateValid, { isDateValid as isDateValidFn } from './isDateValid';
import maxLength, { maxLength as maxLengthFn } from './maxLength';
import minLength, { minLength as minLengthFn } from './minLength';
import notAfterToday, { notAfterToday as notAfterTodayFn } from './notAfterToday';
import notBeforeToday, { notBeforeToday as notBeforeTodayFn } from './notBeforeToday';
import required, { required as requiredFn } from './required';

// Export all rules function
export const rulesFunctions = {
	email: emailFn,
	isDateValid: isDateValidFn,
	maxLength: maxLengthFn,
	minLength: minLengthFn,
	notAfterToday: notAfterTodayFn,
	notBeforeToday: notBeforeTodayFn,
	required: requiredFn
};

// Export all rules
export default {
	email,
	isDateValid,
	maxLength,
	minLength,
	notAfterToday,
	notBeforeToday,
	required
};
