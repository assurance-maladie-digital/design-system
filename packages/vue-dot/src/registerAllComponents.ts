import { VueConstructor } from 'vue';

import { registerComponents, Components } from './helpers/registerComponents';

import { elements } from './elements';
import { patterns } from './patterns';
import { templates } from './templates';

const components: Components[] = [
	elements,
	patterns,
	templates
];

/** Globally register all components (patterns & elements) */
export function registerAllComponents(Vue: VueConstructor): void {
	components.forEach((component) => {
		registerComponents(Vue, component);
	});
}
