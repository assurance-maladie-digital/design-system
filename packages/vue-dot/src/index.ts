import { VueConstructor } from 'vue';

import { directives } from './directives';
import { registerDirectives } from './helpers/registerDirectives';

import { registerAllComponents } from './registerAllComponents';

import { VueDotPlugin } from '../types';

import './styles/index.scss';

const VueDot: VueDotPlugin = {
	install(Vue: VueConstructor, options) {
		registerDirectives(Vue, directives);
		registerAllComponents(Vue);

		if (options) {
			Vue.prototype.$vd = options;
		}
	}
};

/* istanbul ignore if */
if (typeof window !== 'undefined' && window.Vue) {
	window.Vue.use(VueDot);
}

export default VueDot;
