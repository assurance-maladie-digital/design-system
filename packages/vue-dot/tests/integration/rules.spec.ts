import { rules, rulesFunctions } from '../../src/rules';

describe('rules', () => {
	it('should list all rules', () => {
		expect(rules).toMatchSnapshot();
	});

	it('should list all rules functions', () => {
		expect(rulesFunctions).toMatchSnapshot();
	});
});
