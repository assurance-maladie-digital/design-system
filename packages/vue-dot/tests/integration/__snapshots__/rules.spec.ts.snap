// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rules should list all rules 1`] = `
Object {
  "email": [Function],
  "isDateValid": [Function],
  "maxLength": [Function],
  "minLength": [Function],
  "notAfterToday": [Function],
  "notBeforeToday": [Function],
  "required": [Function],
}
`;

exports[`rules should list all rules functions 1`] = `
Object {
  "email": [Function],
  "isDateValid": [Function],
  "maxLength": [Function],
  "minLength": [Function],
  "notAfterToday": [Function],
  "notBeforeToday": [Function],
  "required": [Function],
}
`;
